glean-core/ffi/src/lib.rs

```diff
@@ -303,6 +303,16 @@ pub extern "C" fn glean_clear_application_lifetime_metrics() {
     with_glean_value(|glean| glean.clear_application_lifetime_metrics());
 }
 
+#[no_mangle]
+pub extern "C" fn glean_set_dirty_flag(flag: u8) {
+    with_glean_value_mut(|glean| glean.set_dirty_flag(flag != 0));
+}
+
+#[no_mangle]
+pub extern "C" fn glean_is_dirty_flag_set() -> u8 {
+    with_glean(|glean| Ok(glean.is_dirty_flag_set()))
```

Comment:
```suggestion
    with_glean_value(|glean| glean.is_dirty_flag_set())
```

would be the alternative way to write this (we also do this for `get_upload_enabled`). It's effectively the same code, so your judgement.